stdlib/tools/swift-reflection-test/swift-reflection-test.c

```diff
@@ -772,47 +820,49 @@ int doDumpHeapInstance(const char *BinaryFilename) {
       exit(status);
     }
     default: { // Parent
-      close(PipeMemoryReader_getChildReadFD(&Pipe));
-      close(PipeMemoryReader_getChildWriteFD(&Pipe));
+      close(PipeMemoryReader_getChildReadFD(Reader));
+      close(PipeMemoryReader_getChildWriteFD(Reader));
       SwiftReflectionContextRef RC =
           swift_reflection_createReflectionContextWithDataLayout(
-              (void *)&Pipe, PipeMemoryReader_queryDataLayout,
+              (void *)Reader, PipeMemoryReader_queryDataLayout,
               PipeMemoryReader_freeBytes, PipeMemoryReader_readBytes,
               PipeMemoryReader_getStringLength,
               PipeMemoryReader_getSymbolAddress);
 
-      uint8_t PointerSize = PipeMemoryReader_getPointerSize((void*)&Pipe);
+      uint8_t PointerSize = PipeMemoryReader_getPointerSize((void*)Reader);
       if (PointerSize != sizeof(uintptr_t))
         errorAndExit("Child process had unexpected architecture");
 
 #if defined(__APPLE__) && defined(__MACH__)
-      PipeMemoryReader_receiveImages(RC, &Pipe);
+      PipeMemoryReader_receiveImages(RC, Reader);
 #else
-      PipeMemoryReader_receiveReflectionInfo(RC, &Pipe);
+      PipeMemoryReader_receiveReflectionInfo(RC, Reader);
 #endif
 
       while (1) {
-        InstanceKind Kind = PipeMemoryReader_receiveInstanceKind(&Pipe);
+	// Flush the cache between every reflection operation
+	flushPipeMemoryReader(Reader);
```

Comment:
Important:  We flush the cache between reflection operations.  Otherwise, we can miss changes to target memory that occur between operations.